tests/core/test_util.py

```diff
@@ -83,3 +84,15 @@ def test_azure_url_with_nested_blob():
     assert url.account_name == "my_account"
     assert url.container == "my_container"
     assert url.blob == "a/b/c/d/e/my_blob"
+
+
+def test_gcs_url():
+    url = GCSUrl("gs://my_bucket/my_blob")
+    assert url.bucket == "my_bucket"
+    assert url.blob == "my_blob"
+
+
+def test_gcs_url_with_nested_blob():
+    url = GCSUrl("gs://my_bucket/a/b/c/d/e/my_blob")
+    assert url.bucket == "my_bucket"
+    assert url.blob == "a/b/c/d/e/my_blob"
```

Comment:
Very simple but like my Azure implementation, I'm always wary of blowing something up with RegEx regardless of how trivial the problem might be 💥

Reply:
This was the reason or my comment above. Would it make sense to try some special characters and/or a negative assertion condition here?

Reply:
The URL should always be in the `gs://<BUCKET>/<BLOB>` format as noted by the GCS SDK but I don't think there's any harm in adding additional tests around the two conditions you noted!

Will do that shortly. Thanks!